packages/components/src/focal-point-picker/tooltip/index.native.js

```diff
@@ -67,17 +68,17 @@ function Label( { align, text, xOffset, yOffset } ) {
 
 	useEffect( () => {
 		startAnimation();
-	}, [ visible ] );
+	}, [ visible, startAnimation ] );
 
-	const startAnimation = () => {
+	const startAnimation = useCallback( () => {
```

Comment:
I'd argue that it's better to keep the function directly inside the `useEffect` when possible, for two reasons:

- We can then rely on eslint to fully verify correctness of the deps. Otherwise, someone could remove the `useCallback` wrapper from `startAnimation` without realizing they broke the `useEffect`, for example.
- It's easier to understand when the effect is going to run. You only have to look at one deps array (the one for `useEffect`), instead of looking at multiple deps arrays (`useEffect` + `useCallback`) and synthesizing them in your head. Even in this simple case, it has already confused me — we didn't need the `visible` in the effect's deps array, because it's already in the `useCallback`'s deps array. I didn't notice it until I wrote this comment 😆

I think this should be our default course of action, since the [official docs](https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies) recommend it, and only suggests the `useCallback` workaround as a "last resort".

> I felt the named function improved readability

We can still keep it named if we want, as long as it lives inside the `useEffect` 😄

Reply:
Awesome, thank you for breaking all of that down!